wheel/metadata.py

```diff
@@ -76,7 +83,8 @@ def pkginfo_to_metadata(egg_info_path, pkginfo_path):
         for extra, reqs in sorted(pkg_resources.split_sections(requires),
                                   key=lambda x: x[0] or ''):
             for item in generate_requirements({extra: reqs}):
-                pkg_info[item[0]] = item[1]
+                if item[1] not in pkg_info.get_all(item[0], []):
+                    pkg_info[item[0]] = item[1]
```

Comment:
These lines are probably not necessary?

Reply:
That's incorrect. Without these lines, it's still generating dupes, because `generate_requirements` is called multiple times.

Reply:
Why not just collect everything in a set here, and then add the sorted results to pkg_info[]? It would probably be more efficient than all the 'not in' searches.

Reply:
You could, but I don't think efficiency is a huge concern here.  There will only be a handful of metadata lines, we're not talking about hundreds or thousands of elements to really see the gains from using a set.